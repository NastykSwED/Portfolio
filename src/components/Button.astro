---
import { Icon } from 'astro-icon';

interface Props {
	text: string;
	variant?: string;
	type?: 'button' | 'submit' | 'reset' | undefined;
	iconName: string;
	iconPackName: string;
	iconUtility?: string;
}
const { text, type, variant, iconPackName, iconName, iconUtility } =
	Astro.props;
---

<button class={variant ? `Button Button--${variant}` : 'Button'} type={type}
	>{text}
	<span class={`Button-Icon ${iconUtility}`}>
		<Icon pack={iconPackName} name={iconName} class={iconUtility} />
	</span>
</button>
