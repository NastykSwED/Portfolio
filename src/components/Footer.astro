---
const year = new Date().getFullYear();
---

<footer class='re-Footer' id='re-Footer'>
	<div class='re-Footer-head'>
		<div class='re-Footer-content'>
			<h1 class='re-Footer-title'>Ronald Torres</h1>
			<h2 class='re-Footer-subTitle'>Front End Developer</h2>
		</div>
		<div class='re-Footer-enjoyer'>
			<span>React Enjoyer</span>
		</div>
	</div>
	<div class='re-Footer-legal'>
		<div class='re-Footer-disclaimer'>
			<p class='re-Footer-disclaimerTitle'>
				Â©{year} Ronald Torres. All rights reserved
			</p>
		</div>
	</div>
</footer>
