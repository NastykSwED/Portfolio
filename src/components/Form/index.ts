export * from './ContactFormField';
export * from './ContactFormInput';
export * from './ContactFormTextArea';
