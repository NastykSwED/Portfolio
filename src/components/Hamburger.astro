<div class='re-Hamburger'>
	<span class='re-Hamburger-bar'></span>
	<span class='re-Hamburger-bar'></span>
	<span class='re-Hamburger-bar'></span>
</div>

<script>
	const hamburgerElement = document.querySelector(
		'.re-Hamburger'
	) as HTMLElement;
	const headerNavbarElement = document.querySelector(
		'.re-Navbar'
	) as HTMLElement;

	const handleClick = () => {
		hamburgerElement.classList.toggle('isActive');
		headerNavbarElement.classList.toggle('isActive');
	};

	hamburgerElement.addEventListener('click', handleClick);
</script>
