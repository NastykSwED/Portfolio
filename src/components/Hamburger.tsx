import { useState } from 'react';

interface HamburgerProps {
	onClick: () => void;
}

export const Hamburger = (props: HamburgerProps) => {
	const { onClick } = props;

	const [isActive, setIsActive] = useState<boolean>(false);

	return (
		<div
			data-testid='Hamburger'
			onClick={() => {
				onClick();
				setIsActive(!isActive);
			}}
			className={`re-Hamburger ${isActive ? 'isActive' : ''}`}
		>
			<span className='re-Hamburger-bar'></span>
			<span className='re-Hamburger-bar'></span>
			<span className='re-Hamburger-bar'></span>
		</div>
	);
};
