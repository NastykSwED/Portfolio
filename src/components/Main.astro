---
import ContactSection from './Sections/Contact/ContactSection.astro';
import HomeSection from './Sections/Home/HomeSection.astro';
import ProjectsSection from './Sections/Projects/ProjectsSection.astro';
import SkillsSection from './Sections/Skills/SkillsSection.astro';
---

<main class='re-Main' id='re-Main'>
	<HomeSection />
	<SkillsSection />
	<ProjectsSection />
	<ContactSection />
</main>
