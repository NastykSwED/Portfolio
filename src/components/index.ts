export * from './Buttons';
export * from './Form';
export * from './Loaders';
export * from './Section';
export * from './Navbar/Navbar';
export * from './Hamburger';

