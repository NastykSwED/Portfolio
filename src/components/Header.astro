---
import Hamburger from './Hamburger.astro';
import Navbar from './Navbar/Navbar.astro';
---

<header class='re-Header' id='re-Header'>
	<div class='re-Header-wrapNavbar'>
		<h1 class='re-Header-logo'>Ronald Torres</h1>
		<Navbar />
		<Hamburger />
	</div>
</header>
