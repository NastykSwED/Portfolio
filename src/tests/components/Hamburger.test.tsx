import { describe, it, vi, expect } from 'vitest';

import { render, screen, fireEvent } from '@testing-library/react';

import { Hamburger } from '../../components';

describe('Testing Hamburger Component', () => {
	const mockOnClick = vi.fn();

	it('should match the snapshot', () => {
		const { container } = render(<Hamburger onClick={mockOnClick} />);

		expect(container).toMatchSnapshot();
	});

	it('Clicking on it should add the property active', () => {
		render(<Hamburger onClick={mockOnClick} />);

		const hamburgerContainer = screen.getByTestId('Hamburger') as HTMLElement;

		fireEvent.click(hamburgerContainer);

		expect(mockOnClick).toHaveBeenCalledTimes(1);

		expect(hamburgerContainer.className.includes('isActive')).toBeTruthy();
	});
});
