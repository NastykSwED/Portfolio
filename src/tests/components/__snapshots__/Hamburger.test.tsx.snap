// Vitest Snapshot v1

exports[`Testing Hamburger Component > should match the snapshot 1`] = `
<div>
  <div
    class="re-Hamburger "
    data-testid="Hamburger"
  >
    <span
      class="re-Hamburger-bar"
    />
    <span
      class="re-Hamburger-bar"
    />
    <span
      class="re-Hamburger-bar"
    />
  </div>
</div>
`;
